Src/Base/AMReX_CuptiTrace.H

```diff
@@ -0,0 +1,80 @@
+#ifndef AMREX_CUPTI_TRACE_H_
+#define AMREX_CUPTI_TRACE_H_
+
+#ifdef AMREX_USE_CUPTI
+#include <stdio.h>
+#include <vector>
+#include <cuda.h>
+#include <cupti.h>
+
+// CUPTI buffer size, enough for 4096 activity records in a single buffer;
+// `CUpti_Activity` objects are 8 bytes long
+#define BFR_SIZE (32768)
+
+// 8-byte alignment
+#define ALIGNMENT (8)
+
+// Round down to `align` boundary; mask out last 3 bits (address therefore ends in a 0 or 8)
+#define ALIGN_BFR(bfr, align) ( (uint8_t*) (((uintptr_t)bfr + (align - 1)) & ~ (uintptr_t)(align - 1)) )
+
```

Comment:
Could we move these macros to `AMReX_CuptiTrace.cpp` so that they do not pollute files that include this header?

Reply:
Done, all these macros moved to `AMReX_CuptiTrace.cpp`.